Develop about jsx

rgoodie opened the pull request:
Asking for a review. I moved the container class around in a few spots to allow for the full hero image. This changes the header to full screen, though content is still inside the container and constrained to decent dimensions. Please mostly see work in `localhost:3000/about` for changes. 

After merging this back into `develop` I plan branching off it to build submit-a-sign that @nickttng has built in `-prototype`

kratsg commented on src/layouts/about.jsx:
```diff
@@ -2,7 +2,186 @@ import React from 'react';
+        </div>
+      </div>
+
+      {/* ends conteiner */}
```
typo

kratsg reviewed (commented):
`about.jsx` should be loading from a `team.json` file that contains all of these things, and just loops over. There's a lot of copy/pasting that could be reduced.

`styles-main.css` is somewhat redundant, and should just be called `main.css`.

@nickttng will need to double-check the fonts, as I'm not the expert and can't comment on that.

meltedspork commented on src/layouts/about.jsx:
```diff
@@ -2,7 +2,186 @@ import React from 'react';
+        <h3 className="tracked pb-2">About SignsFive</h3>
+        <p>
+          <strong>
+            Make the search and share for STE(A)M (Science, Technology,
```
Make it one line, no broken line here

rgoodie replied:
`p` and `strong` in the same line?

meltedspork replied:
i meant line 17 and 18 `Make the search and share for STE(A)M (Science, Technology, Engineering, <em>Art</em>, Math) signs easier.`
instead of..
`Make the search and share for STE(A)M (Science, Technology, (break line here)
Engineering, <em>Art</em>, Math) signs easier.`

meltedspork commented on src/layouts/about.jsx:
```diff
@@ -2,7 +2,186 @@ import React from 'react';
+        </p>
+      </section>
+      <div className="row py-5">
+        {/* <!--about-our-story--> */}
```
`about-our-story` can be broken in their own component (jsx)

rgoodie replied:
done

meltedspork commented on src/layouts/about.jsx:
```diff
@@ -2,7 +2,186 @@ import React from 'react';
+
+
+        {/* <!--about-our-name--> */}
+        <div className="col-md-6">
```
`about-our-name` also can be in their own JSX

rgoodie replied:
done

meltedspork commented on src/layouts/about.jsx:
```diff
@@ -2,7 +2,186 @@ import React from 'react';
+          <h3 className="tracked pb-2">Our Name</h3>
+          <p>
+            <strong>
+              You might be wondering, {'"'}What’s the story behind the name?{'"'}
```
its bit confusing reading this from code wise - can we simplify this something like 
```
 <strong>
   {`You might be wondering, "What’s the story behind the name?"`}
</strong>
```
or
```
 <strong>
   You might be wondering, {`"What’s the story behind the name?"`}
</strong>
```

meltedspork commented on src/layouts/about.jsx:
```diff
@@ -2,7 +2,186 @@ import React from 'react';
+        </div>
+      </header>
+
+      <div className="container">
```
This should be its own JSX and also it can be one components and we can use `.map` with it and are it reusable.

rgoodie replied:
Separated out. Team is pulled from data.js file imported into a comp.

rgoodie replied:
main.css renamed

meltedspork commented:
I would like to see if we could import images and fonts with building rather than loading up as static - Webpack provides this. But that is best off as new issue.

meltedspork commented on public/data/people.json:
```diff
@@ -0,0 +1,57 @@
+[{
```
can we rename to `team.json` ?

meltedspork commented on public/data/people.json:
```diff
@@ -0,0 +1,57 @@
+        "full": "Jason Salerno",
+        "status": "DeafBlind",
+        "title": "Lead Product / Fullstack Development",
+        "img": "/images/team-jason.jpg"
```
dont think we need this `img` property - we can reuse by using `id` or `first_name`+ `last_name`.png

meltedspork commented on public/data/people.json:
```diff
@@ -0,0 +1,57 @@
+[{
+        "full": "Jason Salerno",
```
can we do `first_name` and `last_name` ? and add `id`

meltedspork commented on public/data/people.json:
```diff
@@ -0,0 +1,57 @@
+    },
+    {
+        "full": "Melissa Manak",
+        "status": "Deaf / Hard - of -Hearing",
```
Can we change this from `status` to `identify_id` and another to remapped to `identity.json` with identify id?

meltedspork commented on src/layouts/about/our-people.jsx:
```diff
@@ -0,0 +1,7 @@
+import React from 'react';
+
+
+const OurPeople = () => (
```
can we rename this to `Team` ?

meltedspork commented on src/layouts/about/our-team.jsx:
```diff
@@ -0,0 +1,17 @@
+// import React from 'react';
```
@rgoodie - take a look at gist i made - i have made adjust of how we can handle Team and Members components - https://gist.github.com/meltedspork/05350e1ec8cfda1b7ad16914ec63d103

As it follows React's Proptype - see:  https://reactjs.org/docs/typechecking-with-proptypes.html
and also how we set state in constructor - see: https://reactjs.org/docs/state-and-lifecycle.html

Also for data, if we are doing static data -generally we should be using it as json format in flat file, not JS file to manage data and we can easily import json file and that would be bundled with transpiles.

meltedspork commented on src/layouts/about.jsx:
```diff
@@ -1,8 +1,58 @@
+          </strong>
+        </p>
+        <p>
+          In fields where new technologies and technical terms are
```
same here with broken lines

kratsg commented:
Last thing of note is about dealing with fonts. I'm not sure they're being imported correctly...

kratsg commented:
Note: testing failed because of `require.context` which is not easily mockable in jest: https://github.com/facebook/create-react-app/issues/517

meltedspork reviewed (approved):
NICE!!!

meltedspork commented on src/layouts/about/member.jsx:
```diff
@@ -0,0 +1,36 @@
+
+const pathToMemberImage = require.context('../../images/team');
+
+export default class Member extends Component {
```
we can actually make this stateless function similar to submit-a-sign.jsx. but i am fine with either way. 👍

meltedspork commented:
we probably can fix this `require.context` in test with https://github.com/webpack-contrib/mocha-loader instead.

kratsg committed:
about.jsx refactor and squash
- css added in
- data/team.json along with images/team/*.png for people images using require.context for dynamic loading
- evcohen/eslint-plugin-jsx-a11y#339 used for letting eslint know that react-router Link is a11y-compliant
- images are converted to pngs, and resized down to 300pixels to save space where possible
- added tests for team.json and mocked require.context

meltedspork committed:
refactored tests and minor in src
now using chai's expect

meltedspork committed:
remove unnecessary global
